packages/commonwealth/client/scripts/controllers/chain/cosmos/governance-v1.ts

```diff
@@ -0,0 +1,333 @@
+ * gov module version v1. This is considered a patch to make sure v1-enabled chains
+ * load proposals. Eventually we will ideally move back to one governance.ts file.
+ * Patch state:
+ *
+ * - governance.ts uses cosmJS v1beta1 gov
+ * - governance-v1.ts uses telescope-generated v1 gov  */
+
+const stateEnumToString = (status: string): CosmosProposalState => {
+  switch (status) {
+    case 'PROPOSAL_STATUS_UNSPECIFIED':
+      return 'Unspecified';
+    case 'PROPOSAL_STATUS_DEPOSIT_PERIOD':
+      return 'DepositPeriod';
+    case 'PROPOSAL_STATUS_VOTING_PERIOD':
+      return 'VotingPeriod';
+    case 'PROPOSAL_STATUS_PASSED':
+      return 'Passed';
+    case 'PROPOSAL_STATUS_FAILED':
+      return 'Failed';
+    case 'PROPOSAL_STATUS_REJECTED':
+      return 'Rejected';
+    case 'UNRECOGNIZED':
+      return 'Unrecognized';
+    default:
+      throw new Error(`Invalid proposal state: ${status}`);
+  }
+};
+
+const isCompleted = (status: string): boolean => {
+  return status === 'Passed' || status === 'Rejected' || status === 'Failed';
+};
+
+const asciiLiteralToDecimal = async (n: Uint8Array) => {
+  if (!n) return null;
+  // 500000000000000000 = 0.5
+  // dividing by 1000000000000000 gives 3 decimal digits of precision
+  const cosm = await import('@cosmjs/encoding');
+  const nStr = cosm.fromAscii(n);
+  return +new BN(nStr).div(new BN('1000000000000000')) / 1000;
+};
+
+export const marshalTally = (
+  tally: TallyResultSDKType
+): ICosmosProposalTally => {
+  if (!tally) return null;
+  return {
+    yes: new BN(tally.yes_count),
+    abstain: new BN(tally.abstain_count),
+    no: new BN(tally.no_count),
+    noWithVeto: new BN(tally.no_with_veto_count),
+  };
+};
+
+class CosmosGovernanceV1 extends ProposalModule<
+  CosmosApiType,
+  ICosmosProposal,
+  CosmosProposal
+> {
+  private _votingPeriodS: number;
+  private _yesThreshold: number;
+  private _vetoThreshold: number;
+  private _maxDepositPeriodS: number;
+  private _minDeposit: CosmosToken;
+
+  public get vetoThreshold() {
+    return this._vetoThreshold;
+  }
+
+  public get minDeposit() {
+    return this._minDeposit;
+  }
+
+  private _Chain: CosmosChain;
+  private _Accounts: CosmosAccounts;
+
+  public async init(
+    ChainInfo: CosmosChain,
+    Accounts: CosmosAccounts
+  ): Promise<void> {
+    this._Chain = ChainInfo;
+    this._Accounts = Accounts;
+    try {
+      // query chain-wide params
+      const { deposit_params } = await this._Chain.lcd.cosmos.gov.v1.params({
+        paramsType: 'deposit',
+      });
+      const { tally_params } = await this._Chain.lcd.cosmos.gov.v1.params({
+        paramsType: 'tallying',
+      });
+      const { voting_params } = await this._Chain.lcd.cosmos.gov.v1.params({
+        paramsType: 'voting',
+      });
+      this._votingPeriodS = Number(
+        voting_params?.voting_period.replace('s', '')
+      );
+      this._yesThreshold = await Number(tally_params?.threshold);
```

Comment:
No need to await here. Can also use `+` syntax instead of `Number()`, although we may want to use a Big Number library instead if js number overflow is a concern.

Reply:
oops